src/crypto/equihash.cpp

```diff
@@ -303,6 +303,28 @@ bool Equihash<N,K>::BasicSolve(const eh_HashState& base_state,
     return false;
 }
 
+template<size_t MAX_INDICES>
+bool IsProbablyDuplicate(std::shared_ptr<eh_trunc> indices, size_t lenIndices)
+{
+    assert(lenIndices <= MAX_INDICES);
+    bool checked_index[MAX_INDICES] = {false};
+    bool count_checked = 0;
+    for (int z = 0; z < lenIndices; z++) {
+        // Skip over indices we have already paired
+        if (!checked_index[z]) {
+            for (int y = z+1; y < lenIndices; y++) {
+                if (!checked_index[y] && indices.get()[z] == indices.get()[y]) {
+                    // Pair found
```

Comment:
This was intentional. We only want to find pairs, because that indicates a case where we have two identical hash components that XOR to zero. The algorithm in `IsProbablyDuplicate()` will find integer multiples of a pair, because the third occurrence will match with the fourth etc., but it will specifically not trigger on a triple (where we would therefore have one hash occurrence that is not XORing with itself, thus making the set of indices far more likely to be legitimate).